# Twitter Analysis

This script takes the tweets and analyses the Senator's twitter text between the top and bottom centrality ranking groups.  

The following sources were used:


https://developer.twitter.com/en/docs/basics/getting-started

http://docs.tweepy.org/en/v3.5.0/api.html

https://www.pythoncentral.io/introduction-to-tweepy-twitter-for-python/

https://marcobonzanini.com/2015/03/02/mining-twitter-data-with-python-part-1/ https://anaconda.org/conda-forge/tweepy

https://github.com/cjhutto/vaderSentiment#about-the-scoring

In [1]:
#Twitter API
import tweepy
from tweepy import OAuthHandler
 
consumer_key = 'YOUR-CONSUMER-KEY'
consumer_secret = 'YOUR-CONSUMER-SECRET'
access_token = 'YOUR-ACCESS-TOKEN'
access_secret = 'YOUR-ACCESS-SECRET'

auth = OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_secret)
 
api = tweepy.API(auth)

# Senator with the highest centrality rank basic statistics

In [2]:
#most central senator: Christopher Coons - not actively using twitter
#next most central senator: Senator Richard Blumenthal
userID = 'SenBlumenthal'
followers = api.get_user(id=userID).followers_count

fav_ct = 0
rtw_ct = 0
for status in tweepy.Cursor(api.user_timeline, id = userID).items(1000):
    fav_ct += status.favorite_count
    rtw_ct += status.retweet_count

fav_mean = fav_ct / 1000
rt_mean = rtw_ct / 1000

#Recent Activity
import datetime
test_date = datetime.datetime.now() + datetime.timedelta(-30)
 
tweetCount = 0
for status in tweepy.Cursor(api.user_timeline, id = userID).items():
    if (status.created_at > test_date):
        tweetCount = tweetCount + 1
    else:
        break


print ('Popularity Statistics:')
print (userID)
print ('Followers: ' + str(followers))
print ('Favorites average: ' + str(fav_mean))
print ('Retweets average: ' + str(rt_mean))
print ('Number of Tweets in the Past Month: ' + str(tweetCount))


Popularity Statistics:
SenBlumenthal
Followers: 311710
Favorites average: 633
Retweets average: 260
Number of Tweets in the Past Month: 216


# Senator with the lowest centrality rank basic statistics

In [3]:
#least central senator: Richard Shelby
#less fallowers than the highest central senator and not very active on twitter
userID = 'SenShelby'
followers = api.get_user(id=userID).followers_count

fav_ct = 0
rtw_ct = 0
for status in tweepy.Cursor(api.user_timeline, id = userID).items(1000):
    fav_ct += status.favorite_count
    rtw_ct += status.retweet_count

fav_mean = fav_ct / 1000
rt_mean = rtw_ct / 1000

#Recent Activity
import datetime
test_date = datetime.datetime.now() + datetime.timedelta(-30)
 
tweetCount = 0
for status in tweepy.Cursor(api.user_timeline, id = userID).items():
    if (status.created_at > test_date):
        tweetCount = tweetCount + 1
    else:
        break


print ('Popularity Statistics:')
print (userID)
print ('Followers: ' + str(followers))
print ('Favorites average: ' + str(fav_mean))
print ('Retweets average: ' + str(rt_mean))
print ('Number of Tweets in the Past Month: ' + str(tweetCount))


Popularity Statistics:
SenShelby
Followers: 52405
Favorites average: 25
Retweets average: 23
Number of Tweets in the Past Month: 19


# Get Top and Bottom Senators Last 100 Tweets

In [4]:
#Save last 100 tweets twitter data to a csv file
#Tweets were taken from the top 5 senators with highest centrality ranks and from the bottom 5 senators with lowest rank.

import time
import csv
#create loop later, process manually since there are character errors when pulling the data

#top 5 senators
senator="SenCoonsOffice"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets01.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

senator="SenBlumenthal"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets02.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

senator="SenRubioPress"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets03.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

senator="SenFeinstein"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets04.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

senator="SenatorCollins"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets05.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

    
#bottom 5 senators
senator="SenShelby"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets06.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

senator="SenSasse"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets07.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

senator="SenBobCorker"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets08.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

senator="McConnellPress"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets09.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])

senator="SenSanders"
results = api.user_timeline(screen_name=senator, count=100)
csvfile = open('senator_tweets10.csv','wb')
csvwriter = csv.writer(csvfile)
for item in results:
    csvwriter.writerow([unicode(item.user.screen_name).encode("utf-8"),unicode(item.id).encode("utf-8"),unicode(item.created_at).encode("utf-8"),unicode(item.text).encode("utf-8"),unicode(item.retweet_count).encode("utf-8"),unicode(item.favorite_count).encode("utf-8"),item.user.followers_count])


# Top Senator Tweets Words and Sentiment Analysis

In [5]:
#Analyze last 100 tweets
#Read in csv file into panda dataframe

import os
import glob
import pandas as pd
import numpy as np

path = "C:/DataSenators/High"
allFiles = glob.glob(os.path.join(path,"*.csv"))

np_array_list = []
for file_ in allFiles:
    df = pd.read_csv(file_,index_col=None, header=0)
    np_array_list.append(df.as_matrix())

comb_np_array = np.vstack(np_array_list)
df = pd.DataFrame(comb_np_array)
df.columns = ["screen_name", "id", "created_at", "text", "retweet_count", "favorite_count", "followers_count"]
df.head()


screen_name                  id           created_at  \
0  SenCoonsOffice  770626792227045376  2016-08-30 14:18:39   
1  SenCoonsOffice  767828931475075072  2016-08-22 21:00:57   
2  SenCoonsOffice  745719587698786305  2016-06-22 20:46:19   
3  SenCoonsOffice  694583916146065408  2016-02-02 18:11:25   
4  SenCoonsOffice  667093409606344705  2015-11-18 21:33:57   

                                                text retweet_count  \
0  5th Annual Opportunity: Africa conference comi...             4   
1  Sen. Coons to host opioid crisis roundtable w/...             5   
2  Sen. Coons intros bipartisan legislation strea...             7   
3  Sen. Coons announces Class of 2020 United Stat...             3   
4  Sen. Coons' statement on University of Delawar...             3   

  favorite_count followers_count  
0             12            6458  
1              7            6458  
2              9            6458  
3              2            6458  
4              1            6458

In [6]:
#Tweet's data clean-up for analysis
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer

#Remove puntuation
import re
import string
rem = string.punctuation
pattern = r"[{}]".format(rem)

pattern
df['text'] = df['text'].str.replace(pattern, '')

#Remove numerics
#df.text = df.text.str.replace('\d+', '')
df['text'] = df['text'].str.replace('\d+', '')

#lowercase
df['text_lower'] = df['text'].map(lambda x: x.lower())

from nltk.corpus import stopwords
stop = stopwords.words('english')

df['text_without_stopwords'] = df['text_lower'].apply(lambda x: ' '.join([word for word in x.split() if word not in (stop)]))
df.head()

C:\Users\Sonya\AppData\Local\Continuum\Anaconda2\lib\site-packages\nltk\twitter\__init__.py:20: UserWarning: The twython library has not been installed. Some functionality from the twitter package will not be available.
  warnings.warn("The twython library has not been installed. "
C:\Users\Sonya\AppData\Local\Continuum\Anaconda2\lib\site-packages\ipykernel_launcher.py:24: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal


screen_name                  id           created_at  \
0  SenCoonsOffice  770626792227045376  2016-08-30 14:18:39   
1  SenCoonsOffice  767828931475075072  2016-08-22 21:00:57   
2  SenCoonsOffice  745719587698786305  2016-06-22 20:46:19   
3  SenCoonsOffice  694583916146065408  2016-02-02 18:11:25   
4  SenCoonsOffice  667093409606344705  2015-11-18 21:33:57   

                                                text retweet_count  \
0  th Annual Opportunity Africa conference coming...             4   
1  Sen Coons to host opioid crisis roundtable w O...             5   
2  Sen Coons intros bipartisan legislation stream...             7   
3  Sen Coons announces Class of  United States Se...             3   
4  Sen Coons statement on University of Delawares...             3   

  favorite_count followers_count  \
0             12            6458   
1              7            6458   
2              9            6458   
3              2            6458   
4              1            6458   

                                          text_lower  \
0  th annual opportunity africa conference coming...   
1  sen coons to host opioid crisis roundtable w o...   
2  sen coons intros bipartisan legislation stream...   
3  sen coons announces class of  united states se...   
4  sen coons statement on university of delawares...   

                              text_without_stopwords  
0  th annual opportunity africa conference coming...  
1  sen coons host opioid crisis roundtable w ondc...  
2  sen coons intros bipartisan legislation stream...  
3  sen coons announces class united states servic...  
4  sen coons statement university delawares namin...

In [7]:
#Word frequency table
import pandas as pd
import nltk
data = df
top_num = 50

a = data['text_without_stopwords'].str.cat(sep=' ')
words = nltk.tokenize.word_tokenize(a)
word_dist = nltk.FreqDist(words)

print('Top 50 Word Frequencies')
print('='*50)
results=pd.DataFrame(word_dist.most_common(top_num),columns=['Word','Frequency'])

print(results)


Top 50 Word Frequencies
           Word  Frequency
0            rt         76
1         coons         58
2           amp         54
3          bill         48
4           sen         42
5        senate         39
6         netde         38
7             w         35
8       senator         33
9           tax         33
10        today         32
11   chriscoons         30
12         must         25
13         live         25
14           us         24
15      hearing         20
16     congress         20
17        ahead         20
18          act         19
19   bipartisan         18
20       county         17
21        trump         17
22      morning         17
23    committee         17
24         look         17
25        would         16
26          day         16
27   marcorubio         15
28         work         15
29     american         15
30   thomasfire         15
31        state         14
32        house         14
33       health         14
34       reform         14
35  

In [8]:
#Column clean-up
Top_tweets = data.drop(['text','text_lower'],axis=1)
Top_tweets.head()

screen_name                  id           created_at retweet_count  \
0  SenCoonsOffice  770626792227045376  2016-08-30 14:18:39             4   
1  SenCoonsOffice  767828931475075072  2016-08-22 21:00:57             5   
2  SenCoonsOffice  745719587698786305  2016-06-22 20:46:19             7   
3  SenCoonsOffice  694583916146065408  2016-02-02 18:11:25             3   
4  SenCoonsOffice  667093409606344705  2015-11-18 21:33:57             3   

  favorite_count followers_count  \
0             12            6458   
1              7            6458   
2              9            6458   
3              2            6458   
4              1            6458   

                              text_without_stopwords  
0  th annual opportunity africa conference coming...  
1  sen coons host opioid crisis roundtable w ondc...  
2  sen coons intros bipartisan legislation stream...  
3  sen coons announces class united states servic...  
4  sen coons statement university delawares namin...

In [9]:
#Tweet's sentiment score
#Sentiment analyzer object
sid = SentimentIntensityAnalyzer()

#Sentiment Scale
def scale (score):
    if score <= -.2:
        return "negative"
    if score < .2:
        return "neutral"
    if score <= 1:
        return "positive"

                
def tweets_score(tweets):
    sentiment = []
    score = []
    for tweet in tweets:
        ss = sid.polarity_scores(str(tweet))
        sentiment.append(scale(ss['compound']))
        score.append(ss['compound'])   
    return sentiment, score

sentiment, score = tweets_score(Top_tweets.text_without_stopwords)    

Top_tweets['sentiment'] = sentiment
Top_tweets['score'] = score
Top_tweets.head()
#Top_tweets

screen_name                  id           created_at retweet_count  \
0  SenCoonsOffice  770626792227045376  2016-08-30 14:18:39             4   
1  SenCoonsOffice  767828931475075072  2016-08-22 21:00:57             5   
2  SenCoonsOffice  745719587698786305  2016-06-22 20:46:19             7   
3  SenCoonsOffice  694583916146065408  2016-02-02 18:11:25             3   
4  SenCoonsOffice  667093409606344705  2015-11-18 21:33:57             3   

  favorite_count followers_count  \
0             12            6458   
1              7            6458   
2              9            6458   
3              2            6458   
4              1            6458   

                              text_without_stopwords sentiment   score  
0  th annual opportunity africa conference coming...  positive  0.4215  
1  sen coons host opioid crisis roundtable w ondc...  negative -0.6249  
2  sen coons intros bipartisan legislation stream...  positive  0.5994  
3  sen coons announces class united states servic...  positive  0.4215  
4  sen coons statement university delawares namin...   neutral  0.0000

In [10]:
#Frequency count
Top_tweets.groupby('sentiment').count()

screen_name   id  created_at  retweet_count  favorite_count  \
sentiment                                                                
negative           108  108         108            108             108   
neutral            162  162         162            162             162   
positive           224  224         224            224             224   

           followers_count  text_without_stopwords  score  
sentiment                                                  
negative               108                     108    108  
neutral                162                     162    162  
positive               224                     224    224

# Bottom Senator Tweets Words and Sentiment Analysis

In [11]:
#Analyze last 100 tweets
#Read in csv file into panda dataframe

import os
import glob
import pandas as pd
import numpy as np

path = "C:/DataSenators/Low"
allFiles = glob.glob(os.path.join(path,"*.csv"))


np_array_list = []
for file_ in allFiles:
    df = pd.read_csv(file_,index_col=None, header=0)
    np_array_list.append(df.as_matrix())

comb_np_array = np.vstack(np_array_list)
df = pd.DataFrame(comb_np_array)
df.columns = ["screen_name", "id", "created_at", "text", "retweet_count", "favorite_count", "followers_count"]
df.head()


screen_name                  id           created_at  \
0   SenShelby  941001991790219264  2017-12-13 17:48:52   
1   SenShelby  940722216320294912  2017-12-12 23:17:08   
2   SenShelby  940360109510799361  2017-12-11 23:18:15   
3   SenShelby  939835939710558210  2017-12-10 12:35:23   
4   SenShelby  938800281025990656  2017-12-07 16:00:03   

                                                text retweet_count  \
0  The people of Alabama have spoken. Congratulat...           552   
1  I am proud to learn that SEC Chairman Clayton ...             8   
2  Encouraged by the Nat'l Space Council's decisi...            15   
3  I look forward to speaking with @jaketapper th...            36   
4  Looking forward to hearing from @SenatorStrang...            11   

  favorite_count followers_count  
0           2695           52403  
1             18           52403  
2             51           52403  
3            208           52403  
4             25           52403

In [12]:
#Tweet's data clean-up for analysis
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer

#Remove puntuation
import re
import string
rem = string.punctuation
pattern = r"[{}]".format(rem)

pattern
df['text'] = df['text'].str.replace(pattern, '')

#Remove numerics
#df.text = df.text.str.replace('\d+', '')
df['text'] = df['text'].str.replace('\d+', '')

#lowercase
df['text_lower'] = df['text'].map(lambda x: x.lower())

from nltk.corpus import stopwords
stop = stopwords.words('english')

df['text_without_stopwords'] = df['text_lower'].apply(lambda x: ' '.join([word for word in x.split() if word not in (stop)]))
df.head()

C:\Users\Sonya\AppData\Local\Continuum\Anaconda2\lib\site-packages\ipykernel_launcher.py:24: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal


screen_name                  id           created_at  \
0   SenShelby  941001991790219264  2017-12-13 17:48:52   
1   SenShelby  940722216320294912  2017-12-12 23:17:08   
2   SenShelby  940360109510799361  2017-12-11 23:18:15   
3   SenShelby  939835939710558210  2017-12-10 12:35:23   
4   SenShelby  938800281025990656  2017-12-07 16:00:03   

                                                text retweet_count  \
0  The people of Alabama have spoken Congratulati...           552   
1  I am proud to learn that SEC Chairman Clayton ...             8   
2  Encouraged by the Natl Space Councils decision...            15   
3  I look forward to speaking with jaketapper thi...            36   
4  Looking forward to hearing from SenatorStrange...            11   

  favorite_count followers_count  \
0           2695           52403   
1             18           52403   
2             51           52403   
3            208           52403   
4             25           52403   

                                          text_lower  \
0  the people of alabama have spoken congratulati...   
1  i am proud to learn that sec chairman clayton ...   
2  encouraged by the natl space councils decision...   
3  i look forward to speaking with jaketapper thi...   
4  looking forward to hearing from senatorstrange...   

                              text_without_stopwords  
0  people alabama spoken congratulations doug jon...  
1  proud learn sec chairman clayton selected long...  
2  encouraged natl space councils decision sign p...  
3  look forward speaking jaketapper morning cnnso...  
4  looking forward hearing senatorstrange shortly...

In [13]:
#Word frequency table
import pandas as pd
import nltk
data = df
top_num = 50

a = data['text_without_stopwords'].str.cat(sep=' ')
words = nltk.tokenize.word_tokenize(a)
word_dist = nltk.FreqDist(words)

print('Top 50 Word Frequencies')
print('='*50)
results=pd.DataFrame(word_dist.most_common(top_num),columns=['Word','Frequency'])

print(results)

Top 50 Word Frequencies
            Word  Frequency
0             rt         81
1            tax         63
2   senatemajldr         62
3         senate         55
4          today         51
5            amp         44
6      taxreform         37
7           bill         37
8             us         34
9          great         29
10         judge         23
11     statement         23
12          help         21
13       hearing         20
14      kentucky         20
15      american         20
16     mcconnell         20
17          time         19
18          news         18
19      district         18
20    republican         18
21         potus         17
22     president         16
23   republicans         16
24       alabama         16
25        reform         16
26          work         15
27           one         15
28       morning         15
29          need         15
30          want         14
31       support         14
32         proud         14
33         small        

In [14]:
#Column clean-up
Bot_tweets = data.drop(['text','text_lower'],axis=1)
Bot_tweets.head()

screen_name                  id           created_at retweet_count  \
0   SenShelby  941001991790219264  2017-12-13 17:48:52           552   
1   SenShelby  940722216320294912  2017-12-12 23:17:08             8   
2   SenShelby  940360109510799361  2017-12-11 23:18:15            15   
3   SenShelby  939835939710558210  2017-12-10 12:35:23            36   
4   SenShelby  938800281025990656  2017-12-07 16:00:03            11   

  favorite_count followers_count  \
0           2695           52403   
1             18           52403   
2             51           52403   
3            208           52403   
4             25           52403   

                              text_without_stopwords  
0  people alabama spoken congratulations doug jon...  
1  proud learn sec chairman clayton selected long...  
2  encouraged natl space councils decision sign p...  
3  look forward speaking jaketapper morning cnnso...  
4  looking forward hearing senatorstrange shortly...

In [15]:
#Tweet's sentiment score
#Sentiment analyzer object
sid = SentimentIntensityAnalyzer()

#Sentiment Scale
def scale (score):
    if score <= -.5:
        return "negative"
    if score < .5:
        return "neutral"
    if score <= 1:
        return "positive"

                
def tweets_score(tweets):
    sentiment = []
    score = []
    for tweet in tweets:
        ss = sid.polarity_scores(str(tweet))
        sentiment.append(scale(ss['compound']))
        score.append(ss['compound'])   
    return sentiment, score

sentiment, score = tweets_score(Bot_tweets.text_without_stopwords)    

Bot_tweets['sentiment'] = sentiment
Bot_tweets['score'] = score
Bot_tweets.head()
#Bot_tweets

screen_name                  id           created_at retweet_count  \
0   SenShelby  941001991790219264  2017-12-13 17:48:52           552   
1   SenShelby  940722216320294912  2017-12-12 23:17:08             8   
2   SenShelby  940360109510799361  2017-12-11 23:18:15            15   
3   SenShelby  939835939710558210  2017-12-10 12:35:23            36   
4   SenShelby  938800281025990656  2017-12-07 16:00:03            11   

  favorite_count followers_count  \
0           2695           52403   
1             18           52403   
2             51           52403   
3            208           52403   
4             25           52403   

                              text_without_stopwords sentiment   score  
0  people alabama spoken congratulations doug jon...  positive  0.5994  
1  proud learn sec chairman clayton selected long...   neutral  0.4767  
2  encouraged natl space councils decision sign p...   neutral  0.3612  
3  look forward speaking jaketapper morning cnnso...   neutral  0.4019  
4  looking forward hearing senatorstrange shortly...   neutral  0.0000

In [16]:
#Frequency count
Bot_tweets.groupby('sentiment').count()

screen_name   id  created_at  retweet_count  favorite_count  \
sentiment                                                                
negative            27   27          27             27              27   
neutral            311  311         311            310             310   
positive           156  156         156            156             156   

           followers_count  text_without_stopwords  score  
sentiment                                                  
negative                27                      27     27  
neutral                310                     311    311  
positive               156                     156    156

# Conclusion

When comparing one of the top central senators with the very bottom, we see that the top, Senator Richard Blumenthal, has more followers and is more active than the bottom, Senator Richard Shelby.

The top 25 words between the Top and Bottom Senators seem to be similar with words such as "rt", "tax", "bill", "senate", "today" and "american".  The differences in the top 25 words for the Top group is "act", "trump", "committee" and for the Bottom group is "judge", "kentucky" and "news".

The sentiment analysis results for the Top Senators is more positive and shows 224 positive tweets, 162 neutral tweets and 108 negative tweets.  Whereas, the Bottom Senators is extremely neutral and results in 27 negative tweets with 156 positive tweets and 311 neutral tweets.